chrysom/basicClient.go

```diff
@@ -310,7 +309,7 @@ func validateBasicConfig(config *BasicClientConfig) error {
 	}
 
 	if config.Logger == nil {
```

Comment:
This is code smell.  The logger should never be `nil`.  If it is, we should track back to where that's happening and use `sallust.Default()`.

Reply:
I don't see where the `BasicClient.logger` field is actually used.  It looks like it's always getting the logger from the context, which is good.  If it's not being used, then remove this field completely.

However, `getLogger` should never return a `nil`.  If it does, that's a bug.  So, I would remove all of the `nil` checks for the return from this function.  It's just code noise that serves no purpose other than masking bugs.